src/cryptoadvance/spectrum/spectrum.py

```diff
@@ -844,6 +843,46 @@ def getrawchangeaddress(self, wallet, address_type=None):
         # TODO: refill keypool, subscribe
         return desc.getscriptpubkey().address(self.network)
 
+    @walletrpc
+    def getaddressinfo(self, wallet, address: str):
+        ismine = False
+        ischange = False
+        iswatchonly = False
+        scriptpubkey = EmbitScript.from_address(address)
+        sc = Script.query.filter_by(
+            script=scriptpubkey.data.hex(), wallet=wallet
+        ).first()
+        if sc:
+            ismine = sc.index is not None
+        if ismine:
+            ischange = sc.descriptor.internal
+            iswatchonly = sc.descriptor.private_descriptor is not None
```

Comment:
Is this correct?

It is watchonly if there is a private_descriptor? private_descriptor sounds like a descriptor with private keys?

Reply:
Fixed, thanks